### PR TITLE
Canonical URLS

### LINKED ISSUE #978: Add canonical URL to pages
TUM-Live is now available under multiple domains, e.g. live.rbg.tum.de and tum.live. To avoid duplicate/scattered search engine results, we should implement a configurable domain that is set as the website's canonical site in the html header: https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls?hl=de#rel-canonical-link-method

### PR DESCRIPTION
**Pull request by MatthiasReumann**
### Motivation and Context
- Addresses #978 

### Description
- Add canonicalURL to `config.yaml`
- Add struct `CanonicalURL` which provides canonical urls for the start, course, and watch page
- Add `<link rel="canonical" href="..">` to start, course, and watch page

### Steps for Testing
1. Navigate to a start, course, and watch page
2. Open web inspector
3. Check if each `<head>` includes `<link rel="canonical" href="..">` and the URL is correct.
4. 🌟

### CODE REVIEW

File: web/template/watch.gohtml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     {{$stream := .IndexData.TUMLiveContext.Stream}}
     {{$course := .IndexData.TUMLiveContext.Course}}
     <title>{{.IndexData.Branding.Title}} | {{$course.Name}}: {{$stream.Name}}</title>
+    <link rel="canonical" href="{{.IndexData.CanonicalURL.Stream $course.Slug $stream.ID}}" />
```
Comment (joschahenningsen) [inline web/template/watch.gohtml]: This needs to take the version into account (e.g. CAM)

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Looks good so far, can we have this on the info and login pages as well?

File: web/template/watch.gohtml (hunk 1 above)
Comment (MatthiasReumann) [reply web/template/watch.gohtml]: great catch. thanks! 👍

**Review by joschahenningsen — APPROVED** [review]
Thanks!

**Pull request merged**